File: SPCode/UI/Form/LookupForm.vb
Namespace UI
    Public Class LookupForm

        Public Event ItemSelected(ByVal e As SPLookupFormItemSelectedEventArgs)


#Region "Data Members"

        Private _DataSource As SPDataProxy.SPLookupDataTable
        Private Shared _SearchDBConnection As String = String.Empty
#End Region

#Region "Properties"
        Private _SearchType As String = String.Empty
        Public Property SearchType() As String
            Get
                Return _SearchType
            End Get
            Set(ByVal value As String)
                _SearchType = value
            End Set
        End Property

        Private _SubType As String = String.Empty
        Public Property SubType() As String
            Get
                Return _SubType
            End Get
            Set(ByVal value As String)
                _SubType = value
            End Set
        End Property

        Private _SearchResults As DataGridViewSelectedRowCollection = Nothing
        Private _FirstResult As DataGridViewRow = Nothing

        Public ReadOnly Property FirstResult() As DataGridViewRow
            Get
                Return _FirstResult
            End Get
        End Property
        Public ReadOnly Property SearchResults() As DataGridViewSelectedRowCollection
            Get
                Return _SearchResults
            End Get
        End Property
        Public Shared WriteOnly Property SearchDBConnection() As String
            Set(ByVal value As String)
                _SearchDBConnection = value
            End Set
        End Property
#End Region

        Private Sub BindResultList()
            bsSimpleSearch.DataSource = SPSearchHelper.GetSearchResults(SearchType, SubType, SearchFor.Text, SearchText.Text, _SearchDBConnection)
            SimpleResults.DataSource = bsSimpleSearch

            SimpleResults.Columns("ID").Visible = False
            SimpleResults.Columns("Name").AutoSizeMode = DataGridViewAutoSizeColumnMode.Fill
        End Sub
        Private Sub SimpleResults_DoubleClick(ByVal sender As Object, ByVal e As System.EventArgs) Handles SimpleResults.DoubleClick
            If SimpleResults.SelectedRows IsNot Nothing AndAlso SimpleResults.SelectedRows.Count > 0 Then
                Me.DialogResult = System.Windows.Forms.DialogResult.OK
                RaiseEvent ItemSelected(New SPLookupFormItemSelectedEventArgs(SimpleResults.SelectedRows(0), SimpleResults.SelectedRows))

                _FirstResult = SimpleResults.SelectedRows(0)
                _SearchResults = SimpleResults.SelectedRows
            End If
        End Sub
        Private Sub SearchFor_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles SearchFor.SelectedIndexChanged
            If SearchText.CanFocus Then SearchText.Focus()
        End Sub

        Protected Overrides Sub OnLoad(ByVal e As System.EventArgs)
            On Error Resume Next
            Me.DockState = WeifenLuo.WinFormsUI.DockState.DockLeftAutoHide
            SearchFor.SelectedIndex = 0
            BindResultList()

            MyBase.OnLoad(e)
        End Sub
        Protected Overrides Sub OnFormClosing(ByVal e As System.Windows.Forms.FormClosingEventArgs)
            'Me.DialogResult = CType(IIf(e.CloseReason <> CloseReason.UserClosing, System.Windows.Forms.DialogResult.OK, System.Windows.Forms.DialogResult.Cancel), System.Windows.Forms.DialogResult)
            MyBase.OnFormClosing(e)
        End Sub

        Private Sub SearchText_TextChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles SearchText.TextChanged
            BindResultList()
        End Sub
    End Class

    Public Class SPLookupFormItemSelectedEventArgs
        Inherits EventArgs

        Private _SelectedItems As DataGridViewSelectedRowCollection = Nothing
        Private _FirstItem As DataGridViewRow = Nothing

        Public ReadOnly Property FirstItem() As DataGridViewRow
            Get
                Return _FirstItem
            End Get
        End Property
        Public ReadOnly Property SelectedItems() As DataGridViewSelectedRowCollection
            Get
                Return _SelectedItems
            End Get
        End Property

        Public Sub New(ByVal FirstItem As DataGridViewRow, ByVal SelectedItems As DataGridViewSelectedRowCollection)
            Me._FirstItem = FirstItem
            Me._SelectedItems = SelectedItems
        End Sub
    End Class


End Namespace

File: Backup/SoftLogic.Core/Win/UI/Form/LookupForm.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
using SoftLogik.Win.UI.Support;
using SoftLogik.Win.Data;

namespace SoftLogik.Win.UI
	{
		public partial class LookupForm
		{
			public LookupForm()
			{
				InitializeComponent();
			}
			
			public delegate void ItemSelectedEventHandler(SPLookupFormItemSelectedEventArgs e);
			private ItemSelectedEventHandler ItemSelectedEvent;
			
			public event ItemSelectedEventHandler ItemSelected
			{
				add
				{
					ItemSelectedEvent = (ItemSelectedEventHandler) System.Delegate.Combine(ItemSelectedEvent, value);
				}
				remove
				{
					ItemSelectedEvent = (ItemSelectedEventHandler) System.Delegate.Remove(ItemSelectedEvent, value);
				}
			}
			
			
			
			#region Data Members
            private SimpleSearchCollection _DataSource;
			#endregion
			
			#region Properties
			private string _SearchType = string.Empty;
			public string SearchType
			{
				get
				{
					return _SearchType;
				}
				set
				{
					_SearchType = value;
				}
			}
			
			private string _SubType = string.Empty;
			public string SubType
			{
				get
				{
					return _SubType;
				}
				set
				{
					_SubType = value;
				}
			}
			
			private DataGridViewSelectedRowCollection _SearchResults = null;
			private DataGridViewRow _FirstResult = null;
			
			public DataGridViewRow FirstResult
			{
				get
				{
					return _FirstResult;
				}
			}
			public DataGridViewSelectedRowCollection SearchResults
			{
				get
				{
					return _SearchResults;
				}
			}
			#endregion
			
			private void BindResultList()
			{
				bsSimpleSearch.DataSource = SPSearchHelper.GetSearchResults(SearchType, SubType, SearchFor.Text, SearchText.Text);
				SimpleResults.DataSource = bsSimpleSearch;
				
				SimpleResults.Columns["ID"].Visible = false;
				SimpleResults.Columns["Name"].AutoSizeMode = DataGridViewAutoSizeColumnMode.Fill;
			}
			public void SimpleResults_DoubleClick(object sender, System.EventArgs e)
			{
				if ((SimpleResults.SelectedRows != null)&& SimpleResults.SelectedRows.Count > 0)
				{
					this.DialogResult = System.Windows.Forms.DialogResult.OK;
					if (ItemSelectedEvent != null)
						ItemSelectedEvent(new SPLookupFormItemSelectedEventArgs(SimpleResults.SelectedRows[0], SimpleResults.SelectedRows));
					
					_FirstResult = SimpleResults.SelectedRows[0];
					_SearchResults = SimpleResults.SelectedRows;
				}
			}
			public void SearchFor_SelectedIndexChanged(object sender, System.EventArgs e)
			{
				if (SearchText.CanFocus)
				{
					SearchText.Focus();
				}
			}
			
			protected override void OnLoad(System.EventArgs e)
			{
				//On Error Resume Next VBConversions Warning: On Error Resume Next not supported in C#
				this.DockState = WeifenLuo.WinFormsUI.DockState.DockLeftAutoHide;
				SearchFor.SelectedIndex = 0;
				BindResultList();
				
				base.OnLoad(e);
			}
			protected override void OnFormClosing(System.Windows.Forms.FormClosingEventArgs e)
			{
				//Me.DialogResult = CType(IIf(e.CloseReason <> CloseReason.UserClosing, System.Windows.Forms.DialogResult.OK, System.Windows.Forms.DialogResult.Cancel), System.Windows.Forms.DialogResult)
				base.OnFormClosing(e);
			}
			
			public void SearchText_TextChanged(System.Object sender, System.EventArgs e)
			{
				BindResultList();
			}
		}
		
		public class SPLookupFormItemSelectedEventArgs : EventArgs
		{
			
			
			private DataGridViewSelectedRowCollection _SelectedItems = null;
			private DataGridViewRow _FirstItem = null;
			
			public DataGridViewRow FirstItem
			{
				get
				{
					return _FirstItem;
				}
			}
			public DataGridViewSelectedRowCollection SelectedItems
			{
				get
				{
					return _SelectedItems;
				}
			}
			
			public SPLookupFormItemSelectedEventArgs(DataGridViewRow FirstItem, DataGridViewSelectedRowCollection SelectedItems)
			{
				this._FirstItem = FirstItem;
				this._SelectedItems = SelectedItems;
			}
		}
		
		
	}